### PR TITLE
updated BirthdayPage and BirthdaySection to pull data from api, added error and loading state

### PR DESCRIPTION
**Pull request by alocasia28**
Sorry for the multiple commits, ended up with a divergent branch. This PR has updates for BirthdayPage and BirthdaySection alongside minor changes to the use-gifts hook to address a missing parameter, change to gift object mapping logic to match the rest of the pages. 

Gifts from backend with 1 in the category showing on page
image

gift from backend with 1 in the category
Screenshot 2024-03-29 at 9 22 32 pm

### CODE REVIEW

**Comment by Sandz-Studio** [comment]
.env url for backend required to review pr - put in trello for anyone reviewing these pr's to be aware to update

File: src/hooks/use-gifts.js
Comment (Sandz-Studio) [inline src/hooks/use-gifts.js:1]: all good here

**Review by Sandz-Studio — COMMENTED** [review]
import useEffect and const [errorMessage, setErrorMessage] = useState(null); are not being used atm in the birthday section/page, I can't remember now because I'm tired if they will be when we add 404 page not found and loading page? I'm tired.. but can you review in the morning @alocasia28 if not needed then clean up code. Also remove comment as not applicable anymore :)

Other than that the API network req is all good on both the page and component yay!

**Comment by alocasia28** [comment]
Ahh, sorry, had a bit of a merge issue for some reason hence had multiple commits after submission. 

> import useEffect and const [errorMessage, setErrorMessage] = useState(null); are not being used atm in the birthday section/page, I can't remember now because I'm tired if they will be when we add 404 page not found and loading page? I'm tired.. but can you review in the morning @alocasia28 if not needed then clean up code. Also remove comment as not applicable anymore :)
> 
> Other than that the API network req is all good on both the page and component yay!

Reviewed. I want to keep in the useState commented in for now but can delete it later. Comment has been removed.

**Review by Sandz-Studio — APPROVED** [review]

**Pull request merged**